omnigibson/action_primitives/curobo.py

```diff
@@ -0,0 +1,590 @@
+# Third Party
+from collections.abc import Iterable
+
+import numpy as np
+import torch as th  # MUST come before importing omni!!!
+
+import omnigibson as og
+import omnigibson.lazy as lazy
+import omnigibson.utils.transform_utils as T
+from omnigibson.macros import gm
+from omnigibson.utils.control_utils import FKSolver
+
+# Gives 1 - 5% better speedup, according to https://github.com/NVlabs/curobo/discussions/245#discussioncomment-9265692
+th.backends.cudnn.benchmark = True
+th.backends.cuda.matmul.allow_tf32 = True
+th.backends.cudnn.allow_tf32 = True
+
+
+def create_collision_world(tensor_args):
+    """
+    Creates a CuRobo CollisionMeshWorld to use for collision checking
+
+    Args:
+        tensor_args (TensorDeviceType): Tensor device information
+
+    Returns:
+        MeshCollisionWorld: collision world used to check against for collisions
+    """
+    # Checks objA inside objB
+    world = lazy.curobo.geom.types.WorldConfig()
+    usd_help = lazy.curobo.util.usd_helper.UsdHelper()
+    usd_help.stage = og.sim.stage
+    # obstacles = usd_help.get_obstacles_from_stage(only_substring=[objB.prim_path], ignore_substring=[
+    #     "visual"]).get_collision_check_world()  # WorldConfig
+    world_cfg = lazy.curobo.geom.sdf.world.WorldCollisionConfig.load_from_dict(
+        dict(
+            cache={"obb": 10, "mesh": 1024},
+            n_envs=1,
+            checker_type=lazy.curobo.geom.sdf.world.CollisionCheckerType.MESH,
+            max_distance=0.1,
+        ),
+        # obstacles,
+        tensor_args=tensor_args,
+    )
+
+    # To update, run world_coll_checker.load_collision_model(obstacles)
+    return lazy.curobo.geom.sdf.utils.create_collision_checker(world_cfg)
+
+
+def get_obstacles(
+    reference_prim_path=None,
+    only_paths=None,
+    ignore_paths=None,
+    only_substring=None,
+    ignore_substring=None,
+):
+    """
+    Grabs world collision representation
+
+    Args:
+        reference_prim_path (None or str): If specified, the prim path defining the collision world frame
+        only_paths (None or list of str): If specified, only include these sets of prim paths in the resulting
+            collision representation
+        ignore_paths (None or list of str): If specified, exclude these sets of prim paths from the resulting
+            collision representation
+        only_substring (None or list of str): If specified, only include any prim path that includes any of these
+            substrings in the resulting collision representation
+        ignore_paths (None or list of str): If specified, exclude any prim path that includes any of these substrings
+            from the resulting collision representation
+    """
+    usd_help = lazy.curobo.util.usd_helper.UsdHelper()
+    usd_help.stage = og.sim.stage
```

Comment:
Do these USD helper things know to get poses from physx/fabric and not USD?

Reply:
In this PR: https://github.com/StanfordVL/OmniGibson/pull/853, we're trying to always keep flatcache on. Potentially we don't need to test flatcache off here

Reply:
seems to work OK with flatcache on